packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts

```diff
@@ -200,24 +200,57 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
       },
     },
     states: {
-      cameraCheck: {
-        entry: 'resetErrorState',
-        invoke: {
-          src: 'checkVirtualCameraAndGetStream',
-          onDone: {
-            target: 'waitForDOMAndCameraDetails',
-            actions: 'updateVideoMediaStream',
+      initCamera: {
+        initial: 'cameraCheck',
+        states: {
+          cameraCheck: {
+            entry: 'resetErrorState',
+            invoke: {
+              src: 'checkVirtualCameraAndGetStream',
+              onDone: {
+                target: 'waitForDOMAndCameraDetails',
+                actions: 'updateVideoMediaStream',
+              },
+              onError: {
+                target: '#livenessMachine.permissionDenied',
+              },
+            },
+          },
+          waitForDOMAndCameraDetails: {
+            after: {
+              0: {
+                target: '#livenessMachine.initWebsocket',
+                cond: 'hasDOMAndCameraDetails',
+              },
+              10: { target: 'waitForDOMAndCameraDetails' },
+            },
```

Comment:
Any idea how complicated it would be to remove the polling here and have the UI send an update to the machine once the end user has added camera permissions?

Reply:
Let me look into both of these cases to see if we can simplify here :)

Reply:
Was able to remove this polling, good catch!